server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java

```diff
@@ -48,6 +61,141 @@
+
+            Mockito.when(result.getTook()).thenReturn(21L);
+            log.postIndex(ShardId.fromString("[index][123]"), index, result);
+            assertNotNull(appender.getLastEventAndReset());
+        }
+
+        {
+            // level set TRACE, should log when TRACE level is breached
+            settings.updateIndexMetadata(createIndexMetadata(SlowLogLevel.TRACE, "index", uuid));
+            Mockito.when(result.getTook()).thenReturn(10L);
+            log.postIndex(ShardId.fromString("[index][123]"), index, result);
+            assertNull(appender.getLastEventAndReset());
+
+            Mockito.when(result.getTook()).thenReturn(11L);
+            log.postIndex(ShardId.fromString("[index][123]"), index, result);
+            assertNotNull(appender.getLastEventAndReset());
+        }
+    }
+
+    public void testTwoLoggersDifferentLevel() {
+        IndexSettings index1Settings = new IndexSettings(createIndexMetadata(SlowLogLevel.WARN, "index1", UUIDs.randomBase64UUID()),
+            Settings.EMPTY);
+        IndexingSlowLog log1 = new IndexingSlowLog(index1Settings);
+
+        IndexSettings index2Settings = new IndexSettings(createIndexMetadata(SlowLogLevel.TRACE, "index2", UUIDs.randomBase64UUID()),
+            Settings.EMPTY);
+        IndexingSlowLog log2 = new IndexingSlowLog(index2Settings);
+
+
+        ParsedDocument doc = InternalEngineTests.createParsedDoc("1", null);
+        Engine.Index index = new Engine.Index(new Term("_id", Uid.encodeId("doc_id")), randomNonNegativeLong(), doc);
+        Engine.IndexResult result = Mockito.mock(Engine.IndexResult.class);
+        Mockito.when(result.getResultType()).thenReturn(Engine.Result.Type.SUCCESS);
+
+        {
+            // level set WARN, should not log
+            Mockito.when(result.getTook()).thenReturn(11L);
+            log1.postIndex(ShardId.fromString("[index][123]"), index, result);
+            assertNull(appender.getLastEventAndReset());
+
+            // level set TRACE, should log
+            log2.postIndex(ShardId.fromString("[index][123]"), index, result);
+            assertNotNull(appender.getLastEventAndReset());
+        }
+    }
+
+    public void testMultipleSlowLoggersUseSingleLog4jLogger() {
+        LoggerContext context = (LoggerContext) LogManager.getContext(false);
+
+        IndexSettings index1Settings = new IndexSettings(createIndexMetadata(SlowLogLevel.WARN, "index1", UUIDs.randomBase64UUID()),
+            Settings.EMPTY);
+        IndexingSlowLog log1 = new IndexingSlowLog(index1Settings);
+
+        int numberOfLoggersBefore = context.getLoggers().size();
+
+
+        IndexSettings index2Settings = new IndexSettings(createIndexMetadata(SlowLogLevel.TRACE, "index2", UUIDs.randomBase64UUID()),
+            Settings.EMPTY);
+        IndexingSlowLog log2 = new IndexingSlowLog(index2Settings);
+        context = (LoggerContext) LogManager.getContext(false);
+
+        int numberOfLoggersAfter = context.getLoggers().size();
```

Comment:
Maybe we can do this a random number of times, in a loop, say up to 64 or something like that, just to really exercise that no new ones are created?